### PR TITLE
Graphql dgraph

### PR DESCRIPTION
**Pull request by lache-melvin**
## Description

- Updates local devving DGraph Docker image to v23
- Adds a GraphQL schema for Dgraph. There are a few changes related to moving to the GraphQL schema:
  - While Dgraph queries are flexible enough to query across types, GraphQL queries are constrained to a particular type. However DGraph supports nodes having multiple types. 
    - Updated the DataLoader to assign both a generic `Entity` type, as well as the more fine grained type (`Product`/`Route`/`Form` etc) - this allows us to use GraphQL to query the entities across any level of the hierarchy, or target a specific level
  - Removes the `@lang` directive from the `name` predicate - this isn't currently supported through the GraphQL schema (you're supposed to map to language specific fields e.g. `name_en`), but we aren't storing any translations anyway
- Adds npm script to upload the GraphQL schema to Dgraph (this step should probably move into code at some point)
- Adds a new `DgraphGqlDatasource` to the Apollo Server
  - This currently extends a `RESTDataSource`, as it was the quickest way to do a `this.post` of the GraphQL query... next step to replace this with a GraphQL client library, so we get some type support around the queries
- Adds new `entity2` resolver, which resolves an entity as the `entity` resolver does, but using the new `DgraphGqlDatasource`
  - Actually it's slightly less complete than the `entity` resolver - it's currently only returning the data used on the Details page of the existing UC app - to be fleshed out in future PR once we're happy with the pattern
  - There is a `mapEntity` method, as if there are no `properties`, GraphQL returns an empty `properties` array but the existing Dgraph queries return null for some reason (presumably the same for any other array type field)
    - Assuming we need to support the existing API schema, not sure if we just keep to that, doing these mappings, or also create a `v3` that fits more naturally with the GraphQL results?
- To the Admin page of the new frontend (`frontend/packages/host/src/Admin/Settings.tsx`), queries for `entity2` to get a drug and its children/properties (same as in the Details page of the existing Universal Codes app)  - just hardcoded the query for Abacavir :)

## Test

Given that providing a GraphQL schema to Dgraph changes the underlying Dgraph schema, it would be good to double check from a fresh machine that the existing queries are still working as expected (I had many periods along the way where they started returning null, but I think they're ok now 😅)
- Go through [setup instructions](https://github.com/msupply-foundation/unified-codes)
- Test queries (or at least the web app...)
- Upload the GraphQL schema using `npm run update-dgraph-schema`
- Test again

### CODE REVIEW

File: frontend/packages/host/src/api/operations.generated.ts [hunk 1]
```diff
@@ -3,47 +3,83 @@ import * as Types from '@uc-frontend/common';
...
+
+
+export const EntityDocument = gql`
```
[the remark this answers is not in the document]
Comment (lache-melvin) [reply frontend/packages/host/src/api/operations.generated.ts:14]: Yup, the blessing and curse of GraphQL... you only get what you ask for!

[the remark this answers is not in the document]
Comment (lache-melvin) [reply frontend/packages/host/src/api/operations.generated.ts:14]: Put this in :)

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 2]
```diff
@@ -178,15 +178,19 @@ export class DgraphDataSource extends RESTDataSource {
         }
         parents: ~children {
           type: dgraph.type
-          description: name@*
+          description: name
```
Comment (jmbrunskill) [inline unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts:181]: Is this an important change? Any idea what it had @*?

**Review by jmbrunskill — APPROVED** [review]
Looks good to me, had a quick play and things seem to work in both the new and old...

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts (hunk 2 above)
Comment (lache-melvin) [reply unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts:181]: That's the other component of removing the `@lang` directive... `@*` was querying a name in any language

**Pull request merged**